pkg/reconciler/taskrun/validate_taskrun_test.go

```diff
@@ -245,6 +257,62 @@ func TestValidateResolvedTask_InvalidParams(t *testing.T) {
 	}
 }
 
+func TestValidateResolvedTask_InValidExtraParams(t *testing.T) {
+	ctx := context.Background()
+	task := &v1beta1.Task{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+		Spec: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Image:   "myimage",
+				Command: []string{"mycmd"},
+			}},
+			Params: []v1beta1.ParamSpec{
+				{
+					Name: "bar",
+					Type: v1beta1.ParamTypeArray,
+				}, {
+					Name: "include",
+					Type: v1beta1.ParamTypeString,
+				},
+			},
+		},
+	}
+	rtr := &resources.ResolvedTask{
+		TaskSpec: &task.Spec,
+	}
+	p := v1beta1.Params{{
+		Name:  "bar",
+		Value: *v1beta1.NewStructuredValues("somethinggood"),
```

Comment:
Is the reason this is invalid that "bar" should be an array param but is a string param? Can we use table-driven testing for these tests, with descriptive names for each test case, since they test multiple things?

Reply:
Refactored tests to isolate each TaskSpec